.github/workflows/bowline.yml

```
@@ -0,0 +1,158 @@
+        description: 'Run the build with tmate debugging enabled (https://github.com/marketplace/actions/debugging-with-tmate)'
+        required: false
+        default: false
+
+env:
+  GITHUB_PR_NUMBER: ${{github.event.pull_request.number}}
+
+jobs:
+  build-deb-and-tests:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:  [ 'ubuntu-latest' ]
+
+    steps:
+    - name: Setup tmate session
+      uses: mxschmitt/action-tmate@v3
+      if: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.debug_enabled }}
+      with:
+        detached: true
+    - name: env variable
+      run: |
+        echo "deb_version=0.0.0."$(date +%Y%m%d)"" >> $GITHUB_ENV
+        if [ ${{ github.event_name }} == "pull_request" ] ; then
+          echo "head_commit=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+        else
+          echo "head_commit=${{ github.sha }}" >> $GITHUB_ENV
+        fi
+    - uses: actions/checkout@v4
+    - uses: actions/cache@v4
+      id: cache
+      with:
+        path: |
+          cabal-cache
+        key: cache-${{ hashFiles('dnsext-bowline/dnsext-bowline.cabal') }}
+    - name: inject cache into docker
+      uses: reproducible-containers/buildkit-cache-dance@v3.1.0
+      with:
+        cache-map: |
+          {
+            "cabal-cache": "/root/.cabal"
+          }
+        skip-extraction: ${{ steps.cache.outputs.cache-hit }}
+
+    - name: build docker image (and binaries)
+      run: |
+        mkdir -p linux/pkg/opt/bowline/bin linux/pkg/opt/bowline/etc
+        cd build-docker
+        ./build.sh -x ${{ env.head_commit }} ${{ github.server_url }}/${{ github.repository }}
+        cd ..
+        docker run --rm -v /home:/home bowline:bookworm bash -c \
+            "cp /opt/bowline/bin/{bowline,dug,dump} `pwd`/"
+    - name: build deb package
+      run: |
+        mkdir -p pkg/work/bowline-${{ env.deb_version }}/bin
+        docker run --rm -i --name build-bowline \
+            -v /home:/home bowline:bookworm \
+            bash -x -c \
+            "apt-get update ;\
+            apt-get install -y devscripts ;\
+            cd `pwd`; \
+            cd pkg/template; debchange -v ${{ env.deb_version }} -M 'gh action build'; \
+            cd ../work; ln -s ../../../dnsext ./; cd .. ; \
+            ./pkgctl.sh data ${{ env.deb_version }} ;\
+            cp ../{bowline,dug,dump} ./work/bowline-${{ env.deb_version }}/bin/ ;\
+            ./pkgctl.sh pkgs ${{ env.deb_version }} ;\
+            "
+        mv pkg/work/*.deb ./
+        echo "PACKAGE_FILENAME="$(ls *.deb) >> $GITHUB_ENV
+        export PACKAGE_FILENAME=$(ls *.deb)
+        dpkg --contents ${PACKAGE_FILENAME}
+
+    - name: .deb installation
+      run: |
+        docker run -d --name bowline \
+            -p 1853:853/udp -p 1853:853 -p 1443:443/udp -p 1443:443 \
+            -p 1053:53/udp -p 1053:53 -p 1080:80 \
+            -v /home:/home debian:bookworm bash -x -c \
```

```suggestion
            -v /home:/home debian:bookworm-slim bash -x -c \
```

I think the installation tests could also be done with a slim image.

---

ditto;  this can be done but what does it make sense in this context.

https://github.com/debuerreotype/debuerreotype/blob/master/scripts/.slimify-includes

if I looked at the detail of what a `-slim` image is, I don't see a low-hanging fruit here.

---

The intention here is to reuse the image used in "build deb package".